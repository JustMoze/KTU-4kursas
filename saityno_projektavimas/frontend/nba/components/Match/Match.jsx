import React, { useEffect, useState } from 'react'
import styled from 'styled-components'
import { NBA_grey } from '../../styles/globalStyle.module.scss'
import { Row, Col } from 'react-grid-system'
import Modal from '@material-ui/core/Modal'
import { GetUsers } from '../../service/userService'
import Loader from '../Loader/Loader'
import { CalculateWinner } from '../../utils/VictoryAlgorithm'

const CustomModal = styled(Modal)`
  position: absolute;
  background-color: #ffffff;
  z-index: 10;
  width: 70vw;
  height: 80vh;
  left: 15vw !important;
  top: 12vh !important;
  min-height: 500px;
  justify-content: center;
  align-items: center;
  @media (max-width: 768px) {
    width: 94vw;
    left: 3vw !important;
    overflowy: 'scroll';
  }
`
const MatchRow = styled(Row)`
  position: relative;
  background-color: #ffffff;
  height: 100%;
  width: 100%;
  margin-left: 0px !important;
  margin-right: 0px !important;
  overflow: hidden;
`
const MatchCol = styled(Col)`
  height: 100%;
  background-color: ${(props) => props.color};
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  @media (max-width: 768px) {
    height: 50%;
  }
`
const MatchColLeft = styled(MatchCol)`
  border-right: 2px solid black;
  @media (max-width: 768px) {
    border-right: 0px solid black;
  }
`
const RotateSquareUserTeam = styled.div`
  position: absolute;
  margin-left: auto;
  margin-right: auto;
  left: 0;
  right: 0;
  width: 5vw;
  height: 5vw;
  border: 2px solid black;
  transform: rotate(45deg);
  z-index: 2;
  left: 0px;
  background-color: ${(props) => props.color};
  @media (max-width: 1000px) {
    display: none;
  }
`
const MatchButtonContainer = styled.div`
  position: absolute;
  left: 0px;
  right: 0px;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  bottom: 20%;
  @media (max-width: 768px) {
    top: 0px;
    bottom: 0px;
    margin-top: auto;
    margin-bottom: auto;
  }
`
const MatchButton = styled.div`
  width: 10%;
  min-width: 150px;
  border-radius: 20px;
  cursor: pointer;
  min-height: 45px;
  border: 2px solid black;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 7;
  background-color: ${(props) => props.color};
  @media (max-width: 768px) {
  }
`
const LogoContainer = styled.div`
  height: 10vh;
  min-height: 100px;
  width: 100%;
  display: flex;
  position: relative;
  justify-content: center;
  align-items: center;
  @media (max-width: 768px) {
    justify-content: start;
    align-items: center;
  }
`
const LoaderContainer = styled.div`
    width: 100%; 
    height: 100%;
    display: flex; 
    justify-content: center; 
    align-items: center;
`
const LogoDiv = styled.div`
  width: 300px;
  margin-top: 150px;
  @media (max-width: 768px) {
    width: 170px;
    margin-top: 30px;
  }
`
export default function Match({ open, handleClose, team }) {
  const [allPlayers, setAllPlayers] = useState([])
  const [loading, setLoading] = useState(true)
  const [currentPage, setCurrentPage] = useState(1)
  const [prevOpponent, setPrevOpponent] = useState('')
  const [result, setResult] = useState({})
  const [resultLoading, setResultLoading] = useState(true)
  const [battleLoading, setBattleLoading] = useState(false)
  useEffect(() => {
    async function LoadUser() {
      const { data: users } = await GetUsers(currentPage)
      if (users) {
        setAllPlayers(users)
        setLoading(false)
      }
    }
    LoadUser()
    console.log('result in useEffect', result)
  }, [currentPage, result])
  return (
    <CustomModal
      open={open}
      onBackdropClick={handleClose}
      onEscapeKeyDown={handleClose}
      onClose={handleClose}
    >
      <div
        style={{
          backgroundColor: '#ffffff',
          width: '100%',
          height: '100%',
          position: 'relative',
        }}
      >
        {resultLoading ? (
          <MatchRow>
            {battleLoading ? (
              <LoaderContainer><Loader size={25} /></LoaderContainer>
            ) : (
              <>
                <RotateSquareUserTeam color="#aee6e6" style={{ top: '10%' }} />
                <RotateSquareUserTeam
                  color="#aee6e6"
                  style={{ bottom: '10%' }}
                />
                <RotateSquareUserTeam
                  color="#aee6e6"
                  style={{ bottom: '44%' }}
                />
                <MatchColLeft lg={6} md={12} sm={12}>
                  <div
                    style={{
                      width: '100%',
                      height: '95%',
                      position: 'relative',
                    }}
                  >
                    <LogoContainer>
                      <LogoDiv>
                        <img
                          src={`https://images.weserv.nl/?url=https://www.nba.com/assets/logos/teams/primary/web/MIA.svg&w=600&h=600`}
                          alt="1"
                          height="100%"
                          width="100%"
                        />
                      </LogoDiv>
                    </LogoContainer>
                  </div>
                </MatchColLeft>
                <MatchCol
                  lg={6}
                  md={12}
                  sm={12}
                  color="#aee6e6"
                  style={{ zIndex: 6 }}
                >
                  <div
                    style={{
                      width: '100%',
                      height: '95%',
                      position: 'relative',
                      backgroundColor: '#a8dda8',
                      display: 'flex',
                      justifyContent: 'center',
                      alignItems: 'center',
                    }}
                  >
                    <>
                      {loading ? (
                        <LoaderContainer>
                            <Loader size={60} />
                        </LoaderContainer>
                      ) : (
                        <div>
                          <h2>JISAD</h2>
                        </div>
                      )}
                    </>
                  </div>
                </MatchCol>
                <MatchButtonContainer>
                  <MatchButton
                    color="#892cdc"
                    onClick={() => {
                      //setResult(CalculateWinner('5fad20feefc6ac159cfc566d', '5f6b240fd432c92a30ca5159'));
                      // setResultLoading(false);
                      setBattleLoading(true);
                      CalculateWinner(
                        '5fad20feefc6ac159cfc566d',
                        '5f6b240fd432c92a30ca5159'
                      )
                        .then((res) => {
                          setResult(res)
                          setTimeout(() => {
                            setResultLoading(false)
                            setBattleLoading(false)
                          }, 1500)
                        })
                        .catch((err) => {
                          console.log('err', err)
                        })
                    }}
                  >
                    <img
                      src="https://ak-static.cms.nba.com/wp-content/uploads/logos/leagues/logo-nba.svg"
                      height="100%"
                    />
                  </MatchButton>
                </MatchButtonContainer>
              </>
            )}
          </MatchRow>
        ) : (
        <MatchRow>       
                <RotateSquareUserTeam color="#aee6e6" style={{ top: '10%' }} />
                <RotateSquareUserTeam
                  color="#aee6e6"
                  style={{ bottom: '10%' }}
                />
                <RotateSquareUserTeam
                  color="#aee6e6"
                  style={{ bottom: '44%' }}
                />
                <MatchColLeft lg={6} md={12} sm={12}>
                  <div
                    style={{
                      width: '100%',
                      height: '95%',
                      position: 'relative',
                    }}
                  >
                    <LogoContainer>
                      <LogoDiv>
                        <img
                          src={`https://images.weserv.nl/?url=https://www.nba.com/assets/logos/teams/primary/web/MIA.svg&w=600&h=600`}
                          alt="1"
                          height="100%"
                          width="100%"
                        />
                      </LogoDiv>
                    </LogoContainer>
                  </div>
                </MatchColLeft>
                <MatchCol
                  lg={6}
                  md={12}
                  sm={12}
                  color="#aee6e6"
                  style={{ zIndex: 6 }}
                >
                  <div
                    style={{
                      width: '100%',
                      height: '95%',
                      position: 'relative',
                      backgroundColor: '#a8dda8',
                      display: 'flex',
                      justifyContent: 'center',
                      alignItems: 'center',
                    }}
                  >
                        <div>
                             <LoaderContainer>
         <h1>{result.left}  -  {result.right}</h1> 
             </LoaderContainer>
                        </div>
                  </div>
                </MatchCol>
                <MatchButtonContainer>
                  <MatchButton
                    color="#892cdc"
                    onClick={() => {
                      //setResult(CalculateWinner('5fad20feefc6ac159cfc566d', '5f6b240fd432c92a30ca5159'));
                      // setResultLoading(false);
                      setBattleLoading(true);
                      CalculateWinner(
                        '5fad20feefc6ac159cfc566d',
                        '5f6b240fd432c92a30ca5159'
                      )
                        .then((res) => {
                          setResult(res)
                          setTimeout(() => {
                            setResultLoading(false)
                            setBattleLoading(false)
                          }, 1500)
                        })
                        .catch((err) => {
                          console.log('err', err)
                        })
                    }}
                  >
                    <img
                      src="https://ak-static.cms.nba.com/wp-content/uploads/logos/leagues/logo-nba.svg"
                      height="100%"
                    />
                  </MatchButton>
                </MatchButtonContainer>
          </MatchRow>
        )}
      </div>
    </CustomModal>
  )
}
