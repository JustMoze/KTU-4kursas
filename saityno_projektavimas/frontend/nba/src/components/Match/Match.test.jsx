import React from 'react'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import Match, {MatchCol, MatchButton, RotateSquareUserTeam} from './Match'
import { Col } from 'react-grid-system'
import 'jest-styled-components'

configure({ adapter: new Adapter() })

describe('Match component', () => {
    const handleClose = jest.fn();
    const initProps = {
        open: true,
        handleClose: handleClose
    }
   
    it('Renders input correctly', () => {
      const customWrapper = shallow(<Match {...initProps} />)
      expect(toJson(customWrapper)).toMatchSnapshot();
    })
    const wrapper = shallow(<Match {...initProps} />)
    it('Check styled component rendering', () => {
        const MatchButton_div = shallow(<MatchButton color={initProps.color} />)
        const MatchCol_div = shallow(<MatchCol color={initProps.color} />)
        const RotateSquareUserTeam_div = shallow(<RotateSquareUserTeam color={initProps.color} />)
        expect(MatchButton_div.find('div')).toHaveStyleRule('background-color', initProps.color);
        expect(MatchCol_div.find(Col)).toHaveStyleRule('background-color', initProps.color);
        expect(RotateSquareUserTeam_div.find('div')).toHaveStyleRule('background-color', initProps.color);
    })
   
  })