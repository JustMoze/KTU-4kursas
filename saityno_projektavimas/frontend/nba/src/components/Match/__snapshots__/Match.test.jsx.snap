// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match component Renders input correctly 1`] = `
<Match__CustomModal
  onBackdropClick={[MockFunction]}
  onClose={[MockFunction]}
  onEscapeKeyDown={[MockFunction]}
  open={true}
>
  <div
    style={
      Object {
        "backgroundColor": "#ffffff",
        "height": "100%",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Match__MatchRow>
      <Match__RotateSquareUserTeam
        color="#aee6e6"
        style={
          Object {
            "top": "10%",
          }
        }
      />
      <Match__RotateSquareUserTeam
        color="#aee6e6"
        style={
          Object {
            "bottom": "10%",
          }
        }
      />
      <Match__RotateSquareUserTeam
        color="#aee6e6"
        style={
          Object {
            "bottom": "44%",
          }
        }
      />
      <Match__MatchColLeft
        lg={6}
        md={12}
        sm={12}
      >
        <div
          style={
            Object {
              "height": "95%",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <Match__LogoContainer>
            <Match__LogoDiv>
              <img
                alt="1"
                height="100%"
                src="https://images.weserv.nl/?url=https://www.nba.com/assets/logos/teams/primary/web/MIA.svg&w=600&h=600"
                width="100%"
              />
            </Match__LogoDiv>
          </Match__LogoContainer>
        </div>
      </Match__MatchColLeft>
      <Match__MatchCol
        color="#aee6e6"
        lg={6}
        md={12}
        sm={12}
        style={
          Object {
            "zIndex": 6,
          }
        }
      >
        <div
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#a8dda8",
              "display": "flex",
              "height": "95%",
              "justifyContent": "center",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <Match__LoaderContainer>
            <Loader
              size={60}
            />
          </Match__LoaderContainer>
        </div>
      </Match__MatchCol>
      <Match__MatchButtonContainer>
        <Match__MatchButton
          color="#892cdc"
          onClick={[Function]}
        >
          <img
            height="100%"
            src="https://ak-static.cms.nba.com/wp-content/uploads/logos/leagues/logo-nba.svg"
          />
        </Match__MatchButton>
      </Match__MatchButtonContainer>
    </Match__MatchRow>
  </div>
</Match__CustomModal>
`;
